backend/api/communication/src/main/java/co/airy/core/api/communication/SendMessageController.java

```diff
@@ -52,18 +55,24 @@
         String conversationId;
 
         if (payload.getConversationId() != null) {
-            // Append message to existing conversation
-            final ReadOnlyKeyValueStore<String, Conversation> conversationsStore = stores.getConversationsStore();
-            final Conversation conversation = conversationsStore.get(payload.getConversationId().toString());
-            if (conversation == null) {
+            try {
+                Conversation conversation;
+                conversation = sendMessageExecutorService.getFutureCallableConversationAsync(stores, payload).get(5, TimeUnit.SECONDS);
```

Comment:
HTTP call does not return 2 times for single request. For each request there is single response .

Reply:
Yes this is what we talked about if I remember correctly. It should return `202 (accepted)` and in case of error we can use the metadata to notify the caller. Asynchronously